docs/visual-basic/programming-guide/language-features/procedures/extension-methods.md

```diff
@@ -1,181 +1,187 @@
-  
- Extension methods can be declared only within modules. Typically, the module in which an extension method is defined is not the same module as the one in which it is called. Instead, the module that contains the extension method is imported, if it needs to be, to bring it into scope. After the module that contains `Print` is in scope, the method can be called as if it were an ordinary instance method that takes no arguments, such as `ToUpper`:  
-  
- [!code-vb[VbVbalrExtensionMethods#2](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbVbalrExtensionMethods/VB/Class1.vb#2)]  
-  
- The next example, `PrintAndPunctuate`, is also an extension to <xref:System.String>, this time defined with two parameters. The first parameter, `aString`, establishes that the extension method extends <xref:System.String>. The second parameter, `punc`, is intended to be a string of punctuation marks that is passed in as an argument when the method is called. The method displays the string followed by the punctuation marks.  
-  
- [!code-vb[VbVbalrExtensionMethods#3](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbVbalrExtensionMethods/VB/Class2.vb#3)]  
-  
- The method is called by sending in a string argument for `punc`: `example.PrintAndPunctuate(".")`  
-  
- The following example shows `Print` and `PrintAndPunctuate` defined and called. <xref:System.Runtime.CompilerServices> is imported in the definition module in order to enable access to the extension attribute.  
-  
-### Code  
-  
-```vb  
-Imports System.Runtime.CompilerServices  
-  
-Module StringExtensions  
-  
-    <Extension()>   
-    Public Sub Print(ByVal aString As String)  
-        Console.WriteLine(aString)  
-    End Sub  
-  
-    <Extension()>   
-    Public Sub PrintAndPunctuate(ByVal aString As String,   
-                                 ByVal punc As String)  
-        Console.WriteLine(aString & punc)  
-    End Sub  
-  
-End Module  
-```  
-  
- Next, the extension methods are brought into scope and called.  
-  
-```vb  
-Imports ConsoleApplication2.StringExtensions  
-Module Module1  
-  
-    Sub Main()  
-  
-        Dim example As String = "Example string"  
-        example.Print()  
+
+Extension methods enable developers to add custom functionality to data types that are already defined without creating a new derived type. Extension methods make it possible to write a method that can be called as if it were an instance method of the existing type.
   
-        example = "Hello"  
-        example.PrintAndPunctuate(".")  
-        example.PrintAndPunctuate("!!!!")  
+## Remarks
+
+ An extension method can be only a `Sub` procedure or a `Function` procedure. You cannot define an extension property, field, or event. All extension methods must be marked with the extension attribute `<Extension()>` from the <xref:System.Runtime.CompilerServices?displayProperty=nameWithType> namespace.
+
```

Comment:
```suggestion

The `Extension` attribute can only be applied to a Visual Basic [Module](../../../language-reference/statements/module-statement.md). If you apply it to a `Class` or a `Structure`, the Visual Basic compiler generates error [BC36551](../../../misc/bc36551.md), "Extension methods can be defined only in modules."

```

Reply:
> The `Extension` attribute can only be applied to a Visual Basic [Module]

I don't think this one is really accurate. Maybe the following is more accurate:

>  The `Extension` attribute can only be applied to methods defined in a Visual Basic [Module]

Reply:
Well, it can be applied to both. A method defined in a module, and a module.
I've also tested applying it to a class and it generated `BC36550`

`BC36551` is when the method is decorated with Extension attribute, but the method is outside a Module. 

I'll update it.

Reply:
@rpetrusha, Updated it. It's ready for review.